rct229/rulesets/ashrae9012019/section18/section18rule2.py

```diff
@@ -0,0 +1,290 @@
+                    "is_zone_single_zone_sys_b": any(
+                        [
+                            sys_type in APPLICABLE_SYS_TYPES
+                            for sys_type in APPLICABLE_SYS_TYPES
+                        ]
+                    ),
+                    "does_sys_only_serve_lab_b": (
+                        hvac_id_b in lab_zone_hvac_systems["lab_zones_only"]
+                        and lab_zone_exhaust_b > AIRFLOW_15000_CFM
+                        and len(lab_zone_hvac_systems["lab_zones_only"]) == 1
+                    ),
+                    "does_sys_serve_lab_b": (
+                        hvac_id_b in lab_zone_hvac_systems["lab_zones_only"]
+                        and lab_zone_exhaust_b <= AIRFLOW_15000_CFM
+                    ),
+                    "does_sys_serve_lab_and_other_b": (
+                        hvac_id_b in lab_zone_hvac_systems["lab_and_other"]
+                        and lab_zone_exhaust_b <= AIRFLOW_15000_CFM
+                    ),
+                }
+
+                do_multi_zone_evaluation = (
+                    hvac_data_b[hvac_id_b]["is_zone_single_zone_sys_b"]
+                    or not hvac_data_b[hvac_id_b]["does_sys_only_serve_lab_b"]
+                    or not hvac_data_b[hvac_id_b]["does_sys_serve_lab_b"]
+                    or not hvac_data_b[hvac_id_b]["does_sys_serve_lab_and_other_b"]
+                )
+
+                zones_served_by_system = zones_and_terminal_unit_list_dict_b[hvac_id_b][
+                    "zone_list"
+                ]
+
+                zones_on_floor = (
+                    get_zones_on_same_floor_list(
+                        rmd_b,
+                        zones_and_terminal_unit_list_dict_b[hvac_id_b]["zone_list"],
+                    )
+                    if do_multi_zone_evaluation
+                    else []
+                )
+
+                # check if all the zones served by the system are on the same floor
+                if all([zone in zones_on_floor for zone in zones_served_by_system]):
+                    # check if there are any other systems of the same system type that serve zones on this floor
+                    for hvac_system_type2 in applicable_hvac_sys_ids_b:
+                        for hvac_sys2_id_b in baseline_system_types_dict_b[
+                            hvac_system_type2
+                        ]:
+                            # check if hvac_id_b and hvac_sys2_id_b are the same system
+                            if hvac_sys2_id_b != hvac_id_b:
+                                zones_served_by_system2 = (
+                                    zones_and_terminal_unit_list_dict_b[hvac_sys2_id_b][
+                                        "zone_list"
+                                    ]
+                                )
+                                if set(zones_served_by_system2).intersection(
+                                    set(zones_on_floor)
+                                ):
```

Comment:
The condition is whether the length of set will be equal to 0 after intersection.

Reply:
Addressed.